### PR TITLE
feat: support empty operator for verifypermission

### PR DESCRIPTION
**Pull request by jingjunLi**
### Description

Support the need for anonymous users to access public resources

### Rationale

* Anonymous user address is allowed, i.e. operator address is empty
`operator, err := sdk.AccAddressFromHexUnsafe(req.Operator)	if err != nil && err != sdk.ErrEmptyHexAddress`
* The permission in VerifyBucketPermission & VerifyObjectPermission when empty operator. If the resource if public, user can access it.

### Changes

Notable changes: 
* VerifyBucketPermission & VerifyObjectPermission
* ...

### CODE REVIEW

File: x/storage/keeper/query.go [hunk 1]
```diff
@@ -301,7 +301,7 @@ func (k Keeper) VerifyPermission(goCtx context.Context, req *types.QueryVerifyPe
 	ctx := sdk.UnwrapSDKContext(goCtx)
 
 	operator, err := sdk.AccAddressFromHexUnsafe(req.Operator)
-	if err != nil {
+	if err != nil && err != sdk.ErrEmptyHexAddress {
```
Comment (will-2012) [inline x/storage/keeper/query.go:304]: Does only this line of code really affect anonymous users' access to public objects?

**Comment by will-2012** [comment]
What is the difference between v0.1.0 and v0.1.1? Are there compatibility issues?

File: x/storage/keeper/permission.go [hunk 2]
```diff
@@ -82,6 +85,9 @@ func (k Keeper) VerifyObjectPermission(ctx sdk.Context, bucketInfo *types.Bucket
 	if ownerAcc.Equals(operator) {
 		return permtypes.EFFECT_ALLOW
 	}
+	if operator.Empty() {
```
Comment (Fynnss) [inline x/storage/keeper/permission.go:88]: It's wrong. You should put this check to line 83.

BTW, why did your test pass? Please re-check your code.

File: x/storage/keeper/query.go (hunk 1 above)
Comment (Fynnss) [reply x/storage/keeper/query.go:304]: Here just skip the empty address check. The key point is in VerifyObject/BucketPermission

**Review by RenRick — APPROVED** [review]
LGTM

**Comment by Fynnss** [comment]
Pls refine the PR description.

**Pull request merged**